torch/fx/interpreter.py

```diff
@@ -271,6 +272,17 @@ def call_function(self, target : 'Target', args : Tuple[Argument, ...], kwargs :
         """
         assert not isinstance(target, str)
 
+        if not torch._dynamo.config.specialize_float:
+            # Force the specialization of backed symfloats
+            args = tuple(
+                float(a) if isinstance(a, torch.SymFloat) and isinstance(a.node.expr, Symbol) and a.node.hint is not None else a
+                for a in args
+            )
+            kwargs = {
+                k: (float(v) if isinstance(v, torch.SymFloat) and isinstance(v.node.expr, Symbol) and v.node.hint is not None else v)
+                for k, v in kwargs.items()
+            }
```

Comment:
I agree with your analysis in the doc that this is the wrong place to put it. But for mode="eager" it's kind of strange things are failing, because the eager interpreter should be able to handle floats directly, so we don't have to avoid specializing in this case. Hard to have a better idea without more info on the tests that were failing

Reply:
So I put up https://github.com/pytorch/pytorch/pull/138433/files which is basically this diff with the interpreter stuff ripped out.

Here's an example test that fails that diff but passes this one

```
python test/inductor/test_torchinductor.py CpuTests.test_baddbmm_cpu
```

If only there was a way to do set operations on tests across PRs! Then I could quickly get you a list of tests that pass only when this wonky interpreter change is in. I guess it's time to start that side quest ;)

Reply:
I put up https://github.com/pytorch/pytorch/pull/138394 to start logging test failure repros. I didn't know how to test it locally so I just pushed a PR and will check to see if it works in a few hours. Fingers crossed it actually logs correctly!

Reply:
test_cauchy_cuda_float64 looks fun. It sounds like when we started capturing the float we ended up with incorrect semantics in the recorded graph. Shouldn't be difficult to try to solve for directly?